### PR TITLE
Fix some warnings

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: MediaBrowser.Model/Dto/DisplayPreferencesDto.cs [hunk 1]
```diff
@@ -63,10 +63,10 @@ public DisplayPreferencesDto()
         public int PrimaryImageWidth { get; set; }
 
         /// <summary>
-        /// Gets or sets the custom prefs.
+        /// Gets the custom prefs.
         /// </summary>
         /// <value>The custom prefs.</value>
-        public Dictionary<string, string?> CustomPrefs { get; set; }
+        public Dictionary<string, string?> CustomPrefs { get; }
```
Comment (crobibero) [inline MediaBrowser.Model/Dto/DisplayPreferencesDto.cs:69]: I think this will break setting display preferences

Comment (Bond-009) [reply MediaBrowser.Model/Dto/DisplayPreferencesDto.cs]: You're right

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/Library/MediaStreamSelector.cs [hunk 2]
```diff
@@ -13,39 +11,24 @@ namespace Emby.Server.Implementations.Library
...
-                    return defaultStream.Index;
-                }
-            }
-
-            var stream = streams.FirstOrDefault();
-
-            if (stream != null)
-            {
-                return stream.Index;
+                return sortedStreams.FirstOrDefault(i => i.IsDefault)?.Index;
```
Comment (1337joe) [inline Emby.Server.Implementations/Library/MediaStreamSelector.cs:20]: Old behavior: if there is no default stream it falls through and returns the first stream
New behavior: if there is no default stream it returns null, even if there is a non-default stream

Was this intended?

Comment (Bond-009) [reply Emby.Server.Implementations/Library/MediaStreamSelector.cs]: Nice catch, fixed and added test for this case.

**Review by 1337joe — APPROVED** [review]

File: Emby.Server.Implementations/Library/MediaStreamSelector.cs [hunk 3]
```diff
@@ -95,26 +84,27 @@ public static class MediaStreamSelector
             return stream?.Index;
         }
 
-        private static IEnumerable<MediaStream> GetSortedStreams(IEnumerable<MediaStream> streams, MediaStreamType type, string[] languagePreferences)
+        private static IEnumerable<MediaStream> GetSortedStreams(IEnumerable<MediaStream> streams, MediaStreamType type, IReadOnlyList<string> languagePreferences)
         {
             // Give some preference to external text subs for better performance
-            return streams.Where(i => i.Type == type)
+            return streams
+            .Where(i => i.Type == type)
```
Comment (crobibero) [inline Emby.Server.Implementations/Library/MediaStreamSelector.cs:91]:
```suggestion
            return streams
            .Where(i => i.Type == type)
```

Indentation seems off?

**Review by crobibero — APPROVED** [review]

**Pull request merged**